qemu/tests/qmp_command.py

```diff
@@ -200,41 +162,42 @@ def check_result(qmp_o, output=None, exception_list=""):
         qmp_port = qmp_ports[0]
     else:
         test.error("Incorrect configuration, no QMP monitor found.")
-    hmp_ports = vm.get_monitors_by_type('human')
-    if hmp_ports:
-        hmp_port = hmp_ports[0]
-    else:
-        test.error("Incorrect configuration, no QMP monitor found.")
     callback = {"host_cmd": lambda cmd: process.system_output(cmd, shell=True).decode(),
                 "guest_cmd": session.cmd_output,
-                "monitor_cmd": hmp_port.send_args_cmd,
                 "qmp_cmd": qmp_port.send_args_cmd}
 
     def send_cmd(cmd):
-        """ Helper to execute command on ssh/host/monitor """
+        """ Helper to execute command on host/ssh guest/qmp monitor """
```

Comment:
Seems this function is only used to run pre_cmd and post_cmd, right? When it comes to qmp monitor,  `qmp_port.send_args_cmd` is called directly, like L197.  `cmd_type` is gotten from param "event_cmd_type", what does event_cmd mean? It's a bit confusing when several kinds of cmd are used at same time, like below.
>     - qmp_memsave:
>          event_cmd_type = host_cmd
>          qmp_cmd = "memsave val=0, size=4096, filename=/var/tmp/memsave"
>          pre_cmd = "rm -rf /var/tmp/memsave"

Reply:
No,  all of them `pre_cmd`, `post_cmd` and `qmp_cmd` can be executed.
For, `qmp_cmd`, by name, it will be executed with qmp monitor.
And for `pre_cmd` and `post_cmd`, the way to executed it is based on `event_cmd_type`.
All of them commands are needed for the test.

Reply:
I mean it is actually only used for pre_cmd and post_cmd in this script, I don't see anywhere it is used to run qmp cmd.
How about make cmd_type as the function parameter with default value being qmp_cmd, so whenever you want to run a command, you can call this function. 
I guess it's better to explain the event_cmd_type somewhere:)

Reply:
For  first question, every `qmp_cmd` will be executed with:
`197         qmp_o = qmp_port.send_args_cmd(qmp_cmd)` 

For second question,  `cmd_type` is fixed for each case, and it is only used by `send_cmd`,  no other function use it.  If make `cmd_type` as the function (`send_cmd` ) parameter,  The only difference is change L197 (except add and transfer paramter):
from `qmp_o = qmp_port.send_args_cmd(qmp_cmd)` to `pre_o = send_cmd(qmp_cmd, qmp_cmd)`   
So, i think use a global `cmd_type` is enough.
From my view.  `qmp_cmd` is must for every case, and is easy to understand it need be executed with qmp monitor,  I do not want `event_cmd_type` effect `qmp_cmd`.